packages/components/stepper/src/js/stepper.controller.js

```diff
@@ -58,17 +59,20 @@ export default class {
   }
 
   nextStep() {
-    const indexToFocus = Math.min(this.currentIndex + 1, this.steps.length);
-    this.focusStep(indexToFocus);
+    // focusStep() is launch with the $watch
+    this.currentIndex = Math.min(this.currentIndex + 1, this.steps.length);
   }
 
   prevStep() {
-    const indexToFocus = Math.max(this.currentIndex - 1, 0);
-    this.focusStep(indexToFocus);
+    // focusStep() is launch with the $watch
+    this.currentIndex = Math.max(this.currentIndex - 1, 0);
   }
 
   focusStep(indexToFocus) {
-    this.currentIndex = indexToFocus;
+    if (this.currentIndex !== indexToFocus) {
```

Comment:
```suggestion
this.currentIndex = indexToFocus
```

should be enough, except if you do something weird with `this.currentIndex`, then you should probably have a method to update the `currentIndex`

Reply:
It was for the `$watch`, but I'll rewrite it in another way